src/babel-plugin-bare-specifiers.ts

```diff
@@ -29,61 +30,71 @@ type HasSpecifier =
 /**
  * Rewrites so-called "bare module specifiers" to be web-compatible paths.
  */
-export const resolveBareSpecifiers =
-    (filePath: string, isComponentRequest: boolean) => ({
-      inherits: exportExtensions,
-      visitor: {
-        'ImportDeclaration|ExportNamedDeclaration|ExportAllDeclaration'(
-            path: NodePath<HasSpecifier>) {
-          const node = path.node;
+export const resolveBareSpecifiers = (
+    filePath: string,
+    isComponentRequest: boolean,
+    componentDir?: string,
+    rootDir?: string,
+    ) => ({
+  inherits: exportExtensions,
+  visitor: {
+    'ImportDeclaration|ExportNamedDeclaration|ExportAllDeclaration'(
+        path: NodePath<HasSpecifier>) {
+      const node = path.node;
 
-          // An export without a 'from' clause
-          if (node.source == null) {
-            return;
-          }
+      // An export without a 'from' clause
+      if (node.source == null) {
+        return;
+      }
 
-          const specifier = node.source.value;
+      const specifier = node.source.value;
 
-          if (whatwgUrl.parseURL(specifier) !== null) {
-            return;
-          }
+      if (whatwgUrl.parseURL(specifier) !== null) {
+        return;
+      }
 
-          if (isPathSpecifier(specifier)) {
-            return;
-          }
+      if (isPathSpecifier(specifier)) {
+        return;
+      }
 
-          const resolvedSpecifier = resolve.sync(specifier, {
-            basedir: filePath,
-            // Some packages use a non-standard alternative to the "main" field
-            // in their package.json to differentiate their ES module version.
-            packageFilter: (packageJson: {
-              main?: string,
-              module?: string,
-              'jsnext:main'?: string
-            }) => {
+      const resolvedSpecifier = resolve.sync(specifier, {
+        basedir: filePath,
+        // Some packages use a non-standard alternative to the "main" field
+        // in their package.json to differentiate their ES module version.
+        packageFilter:
+            (packageJson:
+                 {main?: string, module?: string, 'jsnext:main'?: string}) => {
               packageJson.main = packageJson.module ||
                   packageJson['jsnext:main'] || packageJson.main;
               return packageJson;
             },
-          });
+      });
 
-          let relativeSpecifierUrl =
-              relative(dirname(filePath), resolvedSpecifier);
+      let relativeSpecifierUrl: string;
+      // If we have an import from that cross the top-level / componentDir
+      // boundary, fix up the path to generate polyserve-style ../ cross-package
+      // imports.
+      if (isComponentRequest && componentDir !== undefined &&
+          rootDir !== undefined && resolvedSpecifier.startsWith(componentDir) &&
```

Comment:
I think startsWith isn't safe for this check unless we ensure there's a trailing slash

Reply:
fixed